### ISSUE TITLE
Add support for classmethod property from python3.9

### ISSUE THREAD

**Issue by YaN-3k** [python/mypy#11619]
```python
class Foo:
    @classmethod
    @property
    def bar(cls) -> str:
        return cls.__name__ 


def foo_bar(msg: str) -> None:
    print(msg)

foo_bar(Foo.bar)
```

> mypy --python-version 3.9 test.py
test.py:11: error: Argument 1 to "foo_bar" has incompatible type "Callable[[], str]"; expected "str"
Found 1 error in 1 file (checked 1 source file)

As stated in https://docs.python.org/3.9/library/functions.html#classmethod this new classmethod property is completely valid so mypy shouldn't report an error.

**Comment by AlexWaygood** [comment]
Might be worth noting that there are [currently active discussions](https://bugs.python.org/issue45356) on whether to deprecate the ability to stack `@classmethod` on top of `@property`, as it's responsible for a _lot_ of bugs in cpython

**Comment by AlexWaygood** [comment]
> It might be worth noting that there are [currently active discussions](https://bugs.python.org/issue45356) on whether to deprecate the ability to stack `@classmethod` on top of `@property`, as it's responsible for a _lot_ of bugs in cpython

To update: this behaviour has now been deprecated in 3.11 (https://github.com/python/cpython/pull/92379)

**Comment by headtr1ck** [comment]
This has been deprecated in python 3.11 it is still correct syntax in python <3.11.
However, it is still semantically incorrent and mypy is right in complaining (the property decorator is not returning what you think it returns).

Maybe one could simply improve the error message for this case?

**Comment by karlicoss** [comment]
Would it be possible to change the error category from `misc` to something more specific? Then it would be easy to suppress it on older python versions
